### PR TITLE
[FREELDR] Ported installfreeldr to Linux.

### PR DESCRIPTION
**Pull request by arnavbhatt288**
Ported installfreeldr to Linux since the finstext2 was obsolete.

- Added new source.
- Removed old source.

### CODE REVIEW

File: boot/freeldr/install/linux/install.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        return -1;
+    }
+
+    if (!OpenVolume(argv[1]))
+    {
+        printf("Exiting program...\n");
+        return -1;
+    }
+    
+	if (!InstallBootSector(argv[2]))
```
Comment (HBelusca) [inline boot/freeldr/install/linux/install.c]: Fix indentation.

File: boot/freeldr/install/linux/install.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
```
Comment (HBelusca) [inline boot/freeldr/install/linux/install.h]: Is this file empty?

File: boot/freeldr/install/linux/volume.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+bool ReadVolumeSector(long SectorNumber, void* SectorBuffer)
+{
+	int dwNumberOfBytesRead;
+	int dwFilePosition;
+	
+	dwFilePosition = lseek(hDiskVolume, (SectorNumber* 512), SEEK_SET);
+	
+	if(dwFilePosition != (SectorNumber * 512))
```
Comment (HBelusca) [inline boot/freeldr/install/linux/volume.c]: `if<space>(...)`

File: boot/freeldr/install/linux/volume.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+bool WriteVolumeSector(long SectorNumber, void* SectorBuffer)
+{
+	int dwNumberOfBytesWritten;
+	int dwFilePosition;
+	
+	dwFilePosition = lseek(hDiskVolume, (SectorNumber * 512), SEEK_SET);
+	
+	if(dwFilePosition != (SectorNumber * 512))
```
Comment (HBelusca) [inline boot/freeldr/install/linux/volume.c]: Ditto

File: boot/freeldr/install/linux/install.c [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        //
+        if (!WriteVolumeSector(14, (fat32_data+512)))
+        {
+            printf("Exiting program...\n");
+            return false;
+        }
+    }
+    else
+    {
+        printf("%s:%d: ", __FILE__, __LINE__);
```
Comment (HBelusca) [inline boot/freeldr/install/linux/install.c]: Why that print that differs from the other ones?

File: boot/freeldr/install/linux/install.c [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        printf("File system type %s unknown.\n", lpszVolumeType);
+        return false;
+    }
+
+    printf("%s boot sector installed.\n", lpszVolumeType);
+
+    return true;
+}
+
+int stricmp(const char *a, const char *b)
```
Comment (HBelusca) [inline boot/freeldr/install/linux/install.c]: This function is already available.

File: boot/freeldr/install/linux/install.h (hunk 2 above)
Comment (arnavbhatt288) [reply boot/freeldr/install/linux/install.h]: Well, the original one was empty.

File: boot/freeldr/install/linux/install.c (hunk 6 above)
Comment (arnavbhatt288) [reply boot/freeldr/install/linux/install.c]: Not really, it is non-standard.

Comment (HBelusca) [reply boot/freeldr/install/linux/install.c]: Then you have the choice of doing something similar to what is already done for the host-tools, namely:
```c
#ifdef _WIN32
#include <string.h>         // Required for _stricmp()
#else
#include <strings.h>        // Required for strcasecmp()
#define _stricmp strcasecmp
#endif
```

Comment (arnavbhatt288) [reply boot/freeldr/install/linux/install.c]: OK, didn't knew that.

**Comment by Extravert-ir** [comment]
Suggestion: adding btrfs support to it would be nice (not in this PR ofc)

**Comment by arnavbhatt288** [comment]
@Extravert-ir I am also adding this file in Cmakelist.txt and testing so it will take some time.

File: boot/freeldr/CMakeLists.txt [hunk 7]
```diff
@@ -3,3 +3,4 @@ add_subdirectory(bootsect)
 add_subdirectory(fdebug)
 add_subdirectory(freeldr)
 add_subdirectory(install)
+
```
Comment (BieHDC) [inline boot/freeldr/CMakeLists.txt]: Accidental change

File: boot/freeldr/install/linux/volume.h [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/*
+ * PROJECT:     ReactOS installfreeldr for linux
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.h]:
```suggestion
 * PROJECT:     ReactOS FreeLoader installer for Linux
```

File: boot/freeldr/install/linux/volume.h [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     installfreeldr for linux.
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.h]:
```suggestion
 * PURPOSE:     Volume functions header file
```

File: boot/freeldr/install/linux/volume.h [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+ * COPYRIGHT:   Copyright 2001  Brian Palmer (brianp@sginet.com)
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.h]:
```suggestion
 * COPYRIGHT:   Copyright 2001 Brian Palmer (brianp@sginet.com)
```

File: boot/freeldr/install/linux/volume.c [hunk 11]
```diff
@@ -0,0 +1,91 @@
+/*
+ * PROJECT:     ReactOS installfreeldr for linux
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.c]:
```suggestion
 * PROJECT:     ReactOS FreeLoader installer for Linux
```

File: boot/freeldr/install/linux/volume.c [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     installfreeldr for linux.
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.c]:
```suggestion
 * PURPOSE:     Volume functions
```

File: boot/freeldr/install/linux/volume.c [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+ * COPYRIGHT:   Copyright 2001  Brian Palmer (brianp@sginet.com)
```
Comment (binarymaster) [inline boot/freeldr/install/linux/volume.c]:
```suggestion
 * COPYRIGHT:   Copyright 2001 Brian Palmer (brianp@sginet.com)
```

File: boot/freeldr/install/linux/install.h [hunk 14]
```diff
@@ -0,0 +1,9 @@
+/*
+ * PROJECT:     ReactOS installfreeldr for linux
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.h]:
```suggestion
 * PROJECT:     ReactOS FreeLoader installer for Linux
```

File: boot/freeldr/install/linux/install.h [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     installfreeldr for linux.
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.h]:
```suggestion
 * PURPOSE:     Installation functions header file
```

File: boot/freeldr/install/linux/install.h [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+ * COPYRIGHT:   Copyright 2001  Brian Palmer (brianp@sginet.com)
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.h]:
```suggestion
 * COPYRIGHT:   Copyright 2001 Brian Palmer (brianp@sginet.com)
```

File: boot/freeldr/install/linux/install.c [hunk 17]
```diff
@@ -0,0 +1,113 @@
+/*
+ * PROJECT:     ReactOS installfreeldr for linux
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.c]:
```suggestion
 * PROJECT:     ReactOS FreeLoader installer for Linux
```

File: boot/freeldr/install/linux/install.c [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     installfreeldr for linux.
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.c]:
```suggestion
 * PURPOSE:     Installation functions
```

File: boot/freeldr/install/linux/install.c [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+ * COPYRIGHT:   Copyright 2001  Brian Palmer (brianp@sginet.com)
```
Comment (binarymaster) [inline boot/freeldr/install/linux/install.c]:
```suggestion
 * COPYRIGHT:   Copyright 2001 Brian Palmer (brianp@sginet.com)
```

**Comment by Extravert-ir** [comment]
Oh, I haven't noticed you have created a completely new version. Can't we use `open`/`lseek`/etc on Windows too?

**Comment by arnavbhatt288** [comment]
@Extravert-ir nope, completely based on Linux.

**Comment by arnavbhatt288** [comment]
Yikes, spelled changing wrong.....

**Comment by Extravert-ir** [comment]
Looks like, you missed my message in the chat, so doubling here:
Thanks for taking a look at our installfreeldr utility, but I'm not going to accept the PR in this state.
You've basically created a second installfreeldr for Linux, which makes no sense. We need a single codebase for this utility.

We may discuss it in chat.reactos.org

**Comment by arnavbhatt288** [comment]
@Extravert-ir Oh well, I don't think this source will work for Windows (only for Linux).
Never mind, I will close the PR and make that repository on my own.

**Pull request closed without merging**